common/src/misc/setLinkedMetadata.decorator.ts

```diff
@@ -0,0 +1,31 @@
+// A decorator used to configure property metadata used by Linked custom fields.
+// Apply it to a class property or getter to make it available as a Linked field option.
+// The id must be unique and must not be changed.
+
+import { ItemView } from '../models/view/itemView';
+
+export class LinkedMetadata {
+    readonly propertyKey: string;
+    private readonly _i18nKey: string;
+
+    constructor(propertyKey: string, i18nKey?: string) {
```

Comment:
Could do a union type of the `keyof` all of the views (snippet below), but that still doesn't solve the thing I actually don't like, which is the cast to `keyof ItemView` in the `cipherView.ts` file.

```typescript 
type viewPropertyKeyType = keyof LoginView | keyof CardView | keyof IdentityView | keyof SecureNoteView;
```

The only protection this would afford is someone manually creating the decorator class would now get an compile-time error on their property name.